aspnetcore/blazor/performance/webassembly-event-pipe-diagnostics.md

```diff
@@ -0,0 +1,122 @@
+
+[!INCLUDE[](~/includes/aspnetcore-repo-ref-source-links.md)]
+
+The following example:
+
+* Collects a GC (Garbage Collector) dump of the live .NET process.
+* Collects diagnostic counters for 60 seconds.
+* Collects CPU counters for 60 seconds.
+
+The MSBuild properties in the following table enable profiler integration.
+
+Property | Default | Set value to&hellip; | Description
+--- | :---: | :---: | ---
+`<WasmPerfTracing>` | `false` | `true` | Controls diagnostic server tracing.
+`<WasmPerfInstrumentation>` | `false` | `true` | Controls CPU sampling instrumentation for diagnostic server. Not necessary for memory dump or counters. **Makes the app execute slower. Only set this to `true` for performance profiling.**
+`<MetricsSupport>` | `false` | `true` | Controls `System.Diagnostics.Metrics` support. For more information, see the [`System.Diagnostics.Metrics` namespace](/dotnet/api/system.diagnostics.metrics).
+`<EventSourceSupport>` | `false`| `true` | Controls `EventPipe` support. For more information, see [Diagnostics and instrumentation: Observability and telemetry](/dotnet/core/deploying/native-aot/diagnostics#observability-and-telemetry).
+
+Enabling profilers has negative size and performance impact, so don't publish an app for production with profilers enabled. In the following example, a condition is set on a property group section that only enables profiling when the app is built with `/p:BlazorSampleProfilingEnabled=true` (.NET CLI) or `<BlazorSampleProfilingEnabled>true</BlazorSampleProfilingEnabled>` in a Visual Studio publish profile, where "`BlazorSampleProfilingEnabled`" is a custom symbol name that you choose and doesn't conflict with other symbol names.
+
+In the app's project file (`.csproj`):
+
+```xml
+<PropertyGroup Condition="'$(BlazorSampleProfilingEnabled)' == 'true'">
+  <WasmPerfTracing>true</WasmPerfTracing>
+  <WasmPerfInstrumentation>true</WasmPerfInstrumentation>
+  <MetricsSupport>true</MetricsSupport>
+  <EventSourceSupport>true</EventSourceSupport>
+</PropertyGroup>
+```
+
+Alternatively, enable features when the app is built with the .NET CLI. The following options passed to the `dotnet build` command mirror the preceding MS Build property configuration:
+
+```dotnetcli
+/p:WasmPerfTracing=true /p:WasmPerfInstrumentation=true /p:MetricsSupport=true /p:EventSourceSupport=true
+```
+
+The [`Timing-Allow-Origin` HTTP header](https://developer.mozilla.org/docs/Web/HTTP/Reference/Headers/Timing-Allow-Origin) allows for more precise time measurements.
+
+Browser developer tools console calls in the following example that trigger profiling:
+
+* `collectGcDump`: Collect a GC (Garbage Collector) dump.
+* `collectPerfCounters(durationSeconds)`: Collect general diagnostic counters.
+* `collectCpuSamples(durationSeconds)`: Collect CPU diagnostic counters.
+
+```javascript
+globalThis.getDotnetRuntime(0).collectGcDump();
```

Comment:
each of these commands is the way how you get the .nettrace, so they could be in each of paragraphs above